sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2219,3 +2203,12 @@ object TimeWindowing extends Rule[LogicalPlan] {
       }
   }
 }
+
+/**
+ * Resolve a [[NamePlaceholder]] into a string [[Literal]].
+ */
+object ResolveNamePlaceholders extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan.transformExpressionsDown {
+    case NamePlaceholder(e) if e.resolved => Literal(e.name)
```

Comment:
Not sure I understand how is it going to be folded during optimization, in any case we need it's actual value during dataType inference (analysis phase)